### PR TITLE
[rocksdb] update to v6.13.3

### PR DESCRIPTION
**Pull request by kobykahane**
- Update port to latest upstream release.
- Rebase patches.
- Fix zstd feature.
- Add missing bzip2 feature.

### CODE REVIEW

File: ports/rocksdb/0001-disable-gtest.patch [hunk 1]
```diff
@@ -1,13 +1,13 @@
...
++++ b/CMakeLists.txt
+@@ -1015,7 +1015,7 @@ endif()
+ option(WITH_ALL_TESTS "Build all test, rather than a small subset" ON)
+ 
+ if(WITH_TESTS OR WITH_BENCHMARK_TOOLS)
+-  add_subdirectory(third-party/gtest-1.8.1/fused-src/gtest)
++  #add_subdirectory(third-party/gtest-1.8.1/fused-src/gtest)
+   add_library(testharness STATIC
+   test_util/mock_time_env.cc
+   test_util/testharness.cc)
```
Comment (NancyLi1013) [inline ports/rocksdb/0001-disable-gtest.patch:13]: The format for this patch seems to be not right. The end of line should be `LF `instead of `CRLF`.

**Review by NancyLi1013 — COMMENTED** [review]
Have you tested these features? Can these work fine in this version?

File: ports/rocksdb/0001-disable-gtest.patch (hunk 1 above)
Comment (kobykahane) [reply ports/rocksdb/0001-disable-gtest.patch]: Fixed.

**Comment by kobykahane** [comment]
> Have you tested these features? Can these work fine in this version?

I've tested consumption of [core,lz4] which is what my downstream project depends on. For the others, I've only tested compilation. In particular, the zstd feature was broken (before this PR) and I've now fixed it. Additionally, a feature for bzip2 was missing and I have now added it as well.

File: ports/rocksdb/0005-backport-msvc-fixes-pr-7439.patch [hunk 2]
```diff
@@ -135,35 +239,18 @@ index 64cdb2f44..514943164 100644
...
  #  define XXH_PREFETCH(ptr)  (void)(ptr)  /* disabled */
  #else
 -#  if defined(_MSC_VER) && (defined(_M_X64) || defined(_M_I86))  /* _mm_prefetch() is not defined outside of x86/x64 */
-+#  if defined(_MSC_VER) && (defined(_M_X64) || defined(_M_IX86))  /* _mm_prefetch() is not defined outside of x86/x64 */
++#if defined(_MSC_VER) && \
++    (defined(_M_X64) ||  \
++     defined(_M_IX86)) /* _mm_prefetch() is not defined outside of x86/x64 */
  #    include <mmintrin.h>   /* https://msdn.microsoft.com/fr-fr/library/84szxsww(v=vs.90).aspx */
  #    define XXH_PREFETCH(ptr)  _mm_prefetch((const char*)(ptr), _MM_HINT_T0)
  #  elif defined(__GNUC__) && ( (__GNUC__ >= 4) || ( (__GNUC__ == 3) && (__GNUC_MINOR__ >= 1) ) )
```
Comment (NancyLi1013) [inline ports/rocksdb/0005-backport-msvc-fixes-pr-7439.patch:256]: There are a lot of changes in this PR. Could you please submit this changes to upstream?

Comment (kobykahane) [reply ports/rocksdb/0005-backport-msvc-fixes-pr-7439.patch:256]: I have already done so and that PR was already merged. This patch is a backport of https://github.com/facebook/rocksdb/pull/7439, which was already merged to upstream master, but is not present in their latest release tag (v6.13.3).

**Review by NancyLi1013 — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks for your contribution!

**Pull request merged**